insights/parsers/krb5.py

```diff
@@ -0,0 +1,152 @@
+and the content format is similar to ``INI config``, but they include
+values that span multiple lines. Multi-line values start with a '{'
+and end with a '}', and we join them together by setting the `is_squ`
+variable to True while in a multi-line value.
+
+The content of file ``/etc/krb5.conf`` and ``/etc/krb5.conf.d/*`` is
+recorded via INI format.
+
+Typical output of the ``/etc/krb5.conf`` is::
+
+Example:
+    >>> krb5_content = '''
+    [realms]
+     dns_lookup_realm = false
+     ticket_lifetime = 24h
+     default_ccache_name = KEYRING:persistent:%{uid}
+     EXAMPLE.COM = {
+      kdc = kerberos.example.com
+      admin_server = kerberos.example.com
+     }
+     89EXAsdcsdMPLE.COM = {
+      kdc = ks2321serberos.example.com
+      admin_server = kc5433scderberos.example.com
+     }
+     [libdefaults]
+      dnsdsd = false
+      tilnvs = 24h
+      89EXAsd = {
+       kdc = ks2321serberos.exampsdcsm
+       admin_server = kc5433scderberos.exasdscdm
+     }
+    # renew_lifetime = 7d
+    # forwardable = true
+    # rdns = false
+    '''.strip()
+
+    >>> from insights.tests import context_wrap
+    >>> shared = {Krb5Configuration: Krb5Configuration(context_wrap(krb5_content))}
+    >>> krb5_info = shared[Krb5Configuration]
+    >>> krb5_info.data["libdefaults"]["dnsdsd"]
+    "false"
+    >>> krb5_info.data["realms"]["EXAMPLE.COM"]["kdc"]
+    "kerberos.example.com"
+"""
+
+
+from insights.core import Parser
+from insights.core.plugins import parser
+from .. import get_active_lines
+
+
+class Krb5ConfigParser(Parser):
+    def parse_content(self, content):
+        dict_all = {}
+        is_squ = False
+        section_name = ""
+        squ_value = {}
+        squ_section_name = ""
+        section_value = {}
+        for line in get_active_lines(content):
+            if "includedir" in line:
+                continue
+            # [XXX] means a section, get the section name and the value format is dict.
+            if line.startswith("[") and line.endswith("]"):
+                # If first section, just get the section name, if not, set the value to the former section and get the section name
+                if section_name:
+                    dict_all[section_name] = section_value
+                section_name = line.strip().strip("[]")
+                section_value = {}
+                continue
+            # Common key value format is XXX = YYY
+            if "=" in line and "{" not in line and "}" not in line and not is_squ:
+                key_value = line.strip().split("=")
+                section_value[key_value[0].strip()] = key_value[1].strip()
+                continue
+            # Special key value format is XXX = YYY{%ZZZ}
+            if "=" in line and "{" in line and "}" in line and len(line.split("=")) == 2 and not is_squ:
+                key_value = line.strip().split("=")
+                section_value[key_value[0].strip()] = key_value[1].strip()
+                continue
```

Comment:
Above two `if` can be merged?
How about:
~~~
if not is_squ and "=" in line and not line.endswith("{"):
~~~